src/components/Input.ts

```diff
@@ -0,0 +1,42 @@
+import { Component, createElement } from "react";
+
+export interface InputProps {
+    disabled: boolean;
+    onChange?: (event: React.ChangeEvent<HTMLInputElement>) => void;
+    color?: string;
+    onKeyDown?: (event: KeyboardEvent) => void;
+}
+
+export class Input extends Component<InputProps, {}> {
+    private colorInputNode!: HTMLInputElement;
+
+    render() {
+        return createElement("div", { className: "widget-color-picker-input-container" },
+            createElement("input", {
+                className: "form-control",
+                type: "text",
+                disabled: this.props.disabled,
+                value: this.props.color,
+                onChange: this.props.onChange,
+                ref: this.getColorInputNodeRef
```

Comment:
Why use `ref` to attache event key listner, cant we use react to do this for you?
https://stackoverflow.com/questions/27827234/keypress-event-handling-in-reactjs

Reply:
Didnt know it could support key events... Thankz